proto/message_contents/proof.proto

```diff
@@ -0,0 +1,32 @@
+// Proof contains signature-based bindings for different use cases.
+syntax = "proto3";
+
+package xmtp.message_contents;
+
+import "message_contents/signature.proto";
+
+option go_package = "github.com/xmtp/proto/v3/go/message_contents";
+option java_package = "org.xmtp.proto.message.contents";
+
+// Association Proof proves that identity key material is associated with
+// a specific individually.
+message AssociationProof {
+    // Legacy: "XMTP : Create Identity ..." Signature Payload
+    message LegacyCreateIdentityProof {
```

Comment:
I was originally hesitant about making these Messages over enums, however in the future messages will make it easier to support 1271